app/models/miq_widget_set.rb

```diff
@@ -1,8 +1,27 @@
 class MiqWidgetSet < ApplicationRecord
+  include_concern 'SetData'
+
   acts_as_miq_set
 
+  before_destroy :ensure_can_be_destroyed
   before_destroy :destroy_user_versions
+  before_destroy :delete_from_dashboard_order
+
+  after_save     :add_to_dashboard_order
   before_save    :keep_group_when_saving
+  after_save     :update_members
+
+  validates :group_id, :presence => true, :unless => ->(x) { x.read_only? }
+
+  validates :name, :format => {:with => /\A[^|]*\z/, :on => :create, :message => "cannot contain \"|\""}
+
+  validate :description do
+    search_params = {:owner_id => group_id, :description => description}
+
+    if description_changed? && MiqWidgetSet.exists?(search_params)
+      errors.add(:error, "Description (Tab Title) must be unique for this group")
+    end
```

Comment:
This feels like a regular uniqueness clause should work when added with a scope on owner_id...

Something like...

```ruby
validates :description, :uniqueness => {:scope => :owner_id}
```

Reply:
good call.

I do notice that this only really does something on description changed
so maybe

```suggestion
validates :description, :uniqueness_when_changed => {:scope => :owner_id}
```